core/citrus-base/src/main/java/org/citrusframework/DefaultTestCase.java

```diff
@@ -237,6 +239,7 @@ public void finish(final TestContext context) {
 
             context.getTestListeners().onTestFinish(this);
             executeFinalActions(context);
+            context.getTestListeners().onFinalActionsEnd(this);
```

Comment:
Instead of introducing a new `onFInalActionsEnd` maybe we can reuse `onTestFinish()` here and just call that after the executeFinalActions?

Reply:
Thought about it but it would be breaking. Therefore the new method. However, for us it does not matter. You are better in judging the consequences.

Reply:
yes, sort of breaking. We can keep the new `onFInalActionsStart` and `onFinalActionsStart` callback methods. 

Still I would prefer to move the `onTestFinish()` method after the final actions execution. It feels more natural IMO as the final actions belong to the test.

If someone needs to run some stuff before the final actions the user can add this to the new `onFinalActionsStart()`.